store/details.go

```diff
@@ -55,3 +58,22 @@ type snapDetails struct {
 	Private     bool   `json:"private"`
 	Confinement string `json:"confinement"`
 }
+
+type snapDeltaDetail struct {
+	FromRevision    int    `json:"from_revision"`
+	ToRevision      int    `json:"to_revision"`
+	Format          string `json:"format"`
+	AnonDownloadURL string `json:"anon_download_url,omitempty"`
+	DownloadURL     string `json:"download_url,omitempty"`
+	Size            int64  `json:"binary_filesize,omitempty"`
+	Sha3_384        string `json:"download_sha3_384,omitempty"`
+}
+
+func getDefaultDetailFields() []string {
+	fields := getStructFields(snapDetails{})
+	// Don't include deltas in the default DetailFields. Instead they'll
+	// be requested explicitly in ListRefresh only.
+	i := sort.SearchStrings(fields, "deltas")
+	fields = append(fields[:i], fields[i+1:]...)
```

Comment:
as we don't really need to preserve order you could instead do
```golang
fields[i] = fields[len(fields)-1]
fields = fields[:len(fields)-1]
```
and avoid the copy